soccer/Context.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include "SystemState.hpp"
+
+struct Context {
+    Context() : state(this) {}
```

Comment:
Delete the copy constructor, copy assign and move constructor to prevent accidental copies?

Reply:
Yeah, these should all be deleted. Good call.